src/components/SidePane/Help/index.tsx

```diff
@@ -1,25 +1,84 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 // eslint-disable-next-line no-restricted-imports
-import {Animated} from 'react-native';
+import {Animated, View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+// @ts-expect-error This is a workaround to display HelpPane on top of everything,
+// Modal from react-native can't be used here, as it would block interactions with the rest of the app
+import ModalPortal from 'react-native-web/dist/exports/Modal/ModalPortal';
+import FocusTrapForModal from '@components/FocusTrap/FocusTrapForModal';
+import SidePaneOverlay from '@components/SidePane/SidePaneOverlay';
 import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
 import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useRootNavigationState from '@hooks/useRootNavigationState';
 import useSafeAreaPaddings from '@hooks/useSafeAreaPaddings';
 import useThemeStyles from '@hooks/useThemeStyles';
 import CONST from '@src/CONST';
+import NAVIGATORS from '@src/NAVIGATORS';
+import ONYXKEYS from '@src/ONYXKEYS';
 import HelpContent from './HelpContent';
 import type HelpProps from './types';
 
-function Help({sidePaneTranslateX, closeSidePane}: HelpProps) {
+function Help({sidePaneTranslateX, closeSidePane, shouldHideSidePaneBackdrop}: HelpProps) {
     const styles = useThemeStyles();
     const {isExtraLargeScreenWidth, shouldUseNarrowLayout} = useResponsiveLayout();
     const {paddingTop, paddingBottom} = useSafeAreaPaddings();
 
-    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ESCAPE, () => closeSidePane(), {isActive: !isExtraLargeScreenWidth});
+    const isInRHP = useRootNavigationState((state) => state?.routes.at(-1)?.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR);
+    const [modal] = useOnyx(ONYXKEYS.MODAL);
+    const isInNarrowPaneModal = !!modal?.isVisible || isInRHP;
+
+    const onCloseSidePaneOnSmallScreens = () => {
+        if (isExtraLargeScreenWidth) {
+            return;
+        }
+
+        closeSidePane();
+    };
+
+    // Close side pane on escape key press
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ESCAPE, () => closeSidePane(), {isActive: !isExtraLargeScreenWidth, shouldBubble: false});
+
+    // Close side pane on small screens when navigation keyboard shortcuts are used
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.SEARCH, onCloseSidePaneOnSmallScreens, {shouldBubble: true});
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.NEW_CHAT, onCloseSidePaneOnSmallScreens, {shouldBubble: true});
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.SHORTCUTS, onCloseSidePaneOnSmallScreens, {shouldBubble: true});
```

Comment:
Logic for hiding the side pane when user triggers navigation shortcuts:

https://github.com/user-attachments/assets/0a9c2c4c-d679-49f7-a2a1-263f219d4f24